app/_src/gateway/production/benchmark.md

```diff
@@ -0,0 +1,218 @@
+The client must also use keep-alive connections. For example, [k6](https://k6.io/docs/using-k6/k6-options/reference/#no-connection-reuse) enables keep-alive by default, and [HTTPClient4](https://hc.apache.org/httpcomponents-client-4.5.x/index.html) implementation in Apache JMeter enables keep-alive by default. Please verify that this is set up appropriately for your test setup.
+
+### Custom plugins
+
+**Action:** Ensure custom plugins aren't interfering with performance.  
+
+Custom plugins can sometimes cause issues with performance. First, you should determine if custom plugins are the source of the performance issues. You can do this by measuring three configuration variations:
+
+1. Measure {{site.base_gateway}}’s performance without enabling any plugins. This provides a baseline for {{site.base_gateway}}’s performance.
+1. Enable necessary bundled plugins (plugins that come with the product), and then measure {{site.base_gateway}}’s performance.
+1. Next, enable custom plugins (in addition to bundled plugins), and then measure {{site.base_gateway}}’s performance once again.
+
+If {{site.base_gateway}}’s baseline performance is poor, then it's likely that either {{site.base_gateway}}’s configuration needs tuning or external factors are affecting it. For external factors, see the other sections in this guide.
+A large difference between the performance in the second and third steps indicates that performance problems could be due to custom plugins.
+
+### Cloud-provider performance issues
+
+**Action:** Ensure you aren't using burstable instances or hitting bandwidth, TCP connection per unit time, or PPS limits. 
+
+While AWS is mentioned in the following, the same recommendations apply to most cloud providers:
+
+* Ensure that you are not using burstable instances, like T type instances, in AWS. In this case, the CPU available to applications is variable, which leads to noise in the stats. For more information, see the [Burstable performance instances](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/burstable-performance-instances.html) AWS documentation.
+* Ensure you are not hitting bandwidth limits, TCP connections per unit time limits, or Packet Per Second (PPS) limits. For more information, see the [Amazon EC2 instance network bandwidth](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-network-bandwidth.html) AWS documentation.
+
+### Configuration changes during benchmark tests
+
+**Action:** Don't change the {{site.base_gateway}} configuration during a benchmark test.
+
+If you change the configuration during a test, {{site.base_gateway}}'s tail latencies can increase sharply. Avoid doing this unless you are measuring {{site.base_gateway}}'s performance under a configuration change.
+
+### Large request and response bodies
+
+**Action:** Keep request bodies below eight KB and response bodies below 32 KB.
```

Comment:
```suggestion
**Action:** Keep request bodies below 8 KB and response bodies below 32 KB.
```
I know we usually spell out a number when it's ten or lower, but I think we should revisit this and set some guidelines on when to use what. When it's a measurement (like KB in this instance), it makes more sense to use the numeral. Additionally, consistency/parallelism is important for clarity, and it's awkward to read "eight KB and 32 KB". 

Interestingly, the [Google style guide](https://developers.google.com/style/numbers#numbers-as-numerals) lists a fair number of exceptions here, including units, and the Mailchimp style guide tells writers to [use the numeral in most situations](https://styleguide.mailchimp.com/grammar-and-mechanics/#header-3-numbers).